### PR TITLE
[Single] Remove gst dependency from single-shot 

### PR DESCRIPTION
**Pull request by gichan-jang**
Remove gst dependency from singlep-shot.

Self evaluation:

Build test: [ *]Passed [ ]Failed [ ]Skipped
Run test: [ *]Passed [ ]Failed [ ]Skipped

Signed-off-by: gichan <gichan2.jang@samsung.com>

### CODE REVIEW

File: packaging/machine-learning-api.spec [hunk 1]
```diff
@@ -297,13 +343,31 @@ cp -r result %{buildroot}%{_datadir}/ml-api/unittest/
...
 
 %files devel
-%{_includedir}/nnstreamer/nnstreamer.h
-%{_includedir}/nnstreamer/nnstreamer-single.h
 %{_libdir}/pkgconfig/capi-ml-inference.pc
 
 %files devel-static
 %{_libdir}/libcapi-nnstreamer.a
 
+%files sinlge
```
Comment (myungjoo) [inline packaging/machine-learning-api.spec:351]: sinlge???

File: packaging/machine-learning-api.spec [hunk 2]
```diff
@@ -297,13 +343,31 @@ cp -r result %{buildroot}%{_datadir}/ml-api/unittest/
...
+%{_libdir}/libcapi-nnstreamer-single.so*
+
+%files sinlge-devel
```
Comment (myungjoo) [inline packaging/machine-learning-api.spec:354]: typo here, too

File: packaging/machine-learning-api.spec [hunk 3]
```diff
@@ -297,13 +343,31 @@ cp -r result %{buildroot}%{_datadir}/ml-api/unittest/
...
+%{_includedir}/nnstreamer/nnstreamer-single.h
+%{_libdir}/pkgconfig/capi-ml-inference-single.pc
+
+%files sinlge-devel-static
```
Comment (myungjoo) [inline packaging/machine-learning-api.spec:358]: single again..

**Comment by myungjoo** [comment]
If we have ml-api-inference-single.so and ml-api-inference-pipeline.so, what does ml-api-inference.so do? What's the dependency relation between the three?

For the purpose of ml-api-inference-single.so (minimal dependency) and the backward compatibility, you might need to make ml-api-inference.so as a dummy .so (w/o actual contents in it) with dependency information only. And the dependency should be "ml-api-inference.so depends on ml-api-inference-single.so & ml-api-inference-pipeline.so" and "ml-api-inference-*.so does NOT depend on the other two".

File: packaging/machine-learning-api.spec (hunk 1 above)
Comment (gichan-jang) [reply packaging/machine-learning-api.spec]: My bad.. Updated all related typo.

**Comment by gichan-jang** [comment]
> If we have ml-api-inference-single.so and ml-api-inference-pipeline.so, what does ml-api-inference.so do? What's the dependency relation between the three?
> 
> For the purpose of ml-api-inference-single.so (minimal dependency) and the backward compatibility, you might need to make ml-api-inference.so as a dummy .so (w/o actual contents in it) with dependency information only. And the dependency should be "ml-api-inference.so depends on ml-api-inference-single.so & ml-api-inference-pipeline.so" and "ml-api-inference-*.so does NOT depend on the other two".

Right. As you mentioned, `capi-nnstreamer.so` is dummy .so link with `capi-nnstreamer-single.so` and `capi-nnstreamer-pipeline.so` for backward compatibility.

**Comment by gichan-jang** [comment]
```
$ ldd libcapi-ml-inference-single.so
	linux-vdso.so.1 (0x00007ffe395fa000)
	libcapi-ml-common.so.1 (0x00007fb37f954000)
	libglib-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0 (0x00007fb37f804000)
	libgmodule-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgmodule-2.0.so.0 (0x00007fb37f7fe000)
	libnnstreamer-single.so (0x00007fb37f7de000)
	libgobject-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0 (0x00007fb37f77e000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fb37f75b000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fb37f567000)
	libpcre.so.3 => /lib/x86_64-linux-gnu/libpcre.so.3 (0x00007fb37f4f4000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fb37f4ee000)
	libffi.so.7 => /usr/lib/x86_64-linux-gnu/libffi.so.7 (0x00007fb37f4e2000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fb37f971000)
```

**Review by jaeyun-jung — APPROVED** [review]

**Pull request merged**